### PR TITLE
add wandb logging, freeze batchnorm by default

### PR DESCRIPTION
**Pull request by mwalmsley**
Doing some polishing on finetuning
- Add wandb logging to the full_tree example. @camallen use this for dashboard. You will need to add import wandb, wandb.init(authkey, etc) just before when running on Azure.
- Freeze batch norm layers by default when finetuning, with new recursive function
- Pass additional params via config (thanks Cam)
- Minor cleanup

### CODE REVIEW

**Comment by mwalmsley** [comment]
Now includes changes from @camallen's param updates also
